ckanext/security/cache/login.py

```diff
@@ -51,24 +47,32 @@ def increment(self):
         # the user would be locked out for another `login_lock_timeout` minutes
         # whenever he/she tries to login again.
         if self.count < self.login_max_count + 1:
-            value.update({self.user_name: "%s:%s" % (self.count + 1, self.request_time)})
-            self.cli.set(self.remote_addr, json.dumps(value))
+            value.update({
+                'count': self.count + 1,
+                'last_failed_attempt': self.request_time,
+            })
+            self.cli.set(self.key, json.dumps(value))
 
-    def needs_lockout(self, cache_value):
-        count, last_attempt = cache_value.split(':')
-        self.count = int(count) if self._check_time(last_attempt) else 0
-        if self._check_count():
-            if self.user is not None and self.count == self.login_max_count:
-                log.info("%s locked out by brute force protection" % self.user.name)
-                try:
-                    notify_lockout(self.user, self.remote_addr)
-                    log.debug("Lockout notification for user %s sent" % self.user.name)
-                except Exception as exc:
-                    msg = "Sending lockout notification for %s failed"
-                    log.exception(msg % self.user.name, exc_info=exc)
-            return False
+    def needs_lockout(self):
+        if self.user is not None and self.count == self.login_max_count:
+            log.info("%s locked out by brute force protection" % self.user.name)
+            try:
+                notify_lockout(self.user)
+                log.debug("Lockout notification for user %s sent" % self.user.name)
+            except Exception as exc:
+                msg = "Sending lockout notification for %s failed"
+                log.exception(msg % self.user.name, exc_info=exc)
+        return False
 
     def check_attempts(self):
-        cached = self.get().get(self.user_name, None)
-        if cached is not None:
-            return self.needs_lockout(cached)
+        value = self.get()
+        in_possible_lockout_window = self._check_time_since_last_attempt(value.get('last_failed_attempt', 0))
+        self.count = value.get('count', 0) if in_possible_lockout_window else 0
```

Comment:
This effectively resets the count if the elapsed time since the last failed attempt is greater than the `login_lock_timeout`